include/api.php

```diff
@@ -3276,72 +3276,706 @@ function api_oauth_access_token($type) {
+		if(!x($_FILES,'image')) {
+			throw new BadRequestException("no media data submitted");
+		}
+
+		// check if specified profile id is valid
+		if ($profileid != 0) {
+			$r = q("SELECT * FROM `profile` WHERE `uid` = %d AND `id` = %d",
+				intval(api_user()),
+				intval($profileid));
+			// error message if specified profile id is not in database
+			if (!dbm::is_result($r))
+				throw new BadRequestException("profile_id not available");
+			$is_default_profile = $r['profile'];
+		} else {
+			$is_default_profile = 1;
+		}
+
+		// get mediadata from image or media (Twitter call api/account/update_profile_image provides image)
+		$media = null;
+		if(x($_FILES, 'image'))
+			$media = $_FILES['image'];
+		elseif(x($_FILES, 'media'))
+			$media = $_FILES['media'];
+
+		// save new profile image
+		$data = SaveMediaToDatabase("profileimage", $media, $type, t('Profile Photos'), "", "", "", "", "", $is_default_profile);
+
+		// get filetype
+		if (is_array($media['type']))
+			$filetype = $media['type'][0];
+		else
+			$filetype = $media['type'];
+
+		if ($filetype = "image/jpeg")
+			$fileext = "jpg";
+		else if ($filetype = "image/png")
+			$fileext = "png";
+
+		// change specified profile or all profiles to the new resource-id
+		if($is_default_profile) {
+			$r = q("UPDATE `photo` SET `profile` = 0 WHERE `profile` = 1 AND `resource-id` != '%s' AND `uid` = %d",
+				dbesc($data['photo']['id']),
+				intval(local_user())
+			);
+
+			$r = q("UPDATE `contact` SET `photo` = '%s', `thumb` = '%s', `micro` = '%s'  WHERE `self` AND `uid` = %d",
+				dbesc(App::get_baseurl() . '/photo/' . $data['photo']['id'] . '-4.' . $fileext),
+				dbesc(App::get_baseurl() . '/photo/' . $data['photo']['id'] . '-5.' . $fileext),
+				dbesc(App::get_baseurl() . '/photo/' . $data['photo']['id'] . '-6.' . $fileext),
+				intval(local_user())
+			);
+		} else {
+			$r = q("update profile set photo = '%s', thumb = '%s' where id = %d and uid = %d",
+				dbesc(App::get_baseurl() . '/photo/' . $data['photo']['id'] . '-4.' . $filetype),
+				dbesc(App::get_baseurl() . '/photo/' . $data['photo']['id'] . '-5.' . $filetype),
+				intval($_REQUEST['profile']),
+				intval(local_user())
+					);
+		}
+
+		// we'll set the updated profile-photo timestamp even if it isn't the default profile,
+		// so that browsers will do a cache update unconditionally
+
+		$r = q("UPDATE `contact` SET `avatar-date` = '%s' WHERE `self` = 1 AND `uid` = %d",
+			dbesc(datetime_convert()),
+			intval(local_user())
+		);
+
+		// Update global directory in background
+		//$user = api_get_user(get_app());
+		$url = App::get_baseurl() . '/profile/' . get_app()->user['nickname'];
+		if ($url && strlen(get_config('system','directory'))) {
+			proc_run(PRIORITY_LOW, "include/directory.php", $url);
+		}
+
+		require_once('include/profile_update.php');
+		profile_change();
+
+		// output for client
+		if ($data) {
+			return api_account_verify_credentials($type);
+		}
+		else {
```

Comment:
Standards: Please move the `else` on the same line as the preceding closing brace.